### PR TITLE
Adding MySQL

### PR DESCRIPTION
**Pull request by abhilash-keloth**
Removed AD and added all other review feedbacks.

### CODE REVIEW

File: examples/mysql_server/100-simple-mysql/configuration.tfvars [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    mysql_configuration = {
+      mysql_configuration = {
+        name                = "interactive_timeout"
+        resource_group_name = "mysql_region1"
+        server_name         = "sales-re1"
+        value               = "600"
+      }
+    }
+
+    mysql_database = {
```
Comment (LaurentLesle) [inline examples/mysql_server/100-simple-mysql/configuration.tfvars:55]: to be consistent with other services can you put a plural 'mysql_databases'

File: examples/mysql_server/100-simple-mysql/configuration.tfvars [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      mysql-firewall-rules = {
+        name = "mysql_server_firewallrule"
+        resource_group_name = "mysql_region1"
+        server_name         = "sales-rg1"
+        start_ip_address    = "10.0.0.1"
+        end_ip_address      = "10.0.0.3"
+      }
+    }
+
+    mysql_configuration = {
```
Comment (LaurentLesle) [inline examples/mysql_server/100-simple-mysql/configuration.tfvars:46]: o be consistent with other services can you put a plural 'mysql_configurations'

File: examples/mysql_server/101-vnet-rule-mysql/configuration.tfvars [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      mysql-firewall-rules = {
+        name = "mysql_server_firewallrule"
+        resource_group_name = "mysql_region1"
+        server_name         = "sales-rg1"
+        start_ip_address    = "10.0.0.1"
+        end_ip_address      = "10.0.0.3"
+      }
+    }
+
+    mysql_configuration = {
```
Comment (LaurentLesle) [inline examples/mysql_server/101-vnet-rule-mysql/configuration.tfvars:50]: plural

File: examples/mysql_server/101-vnet-rule-mysql/configuration.tfvars [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+      }
+    }
+    
+    mysql_vnet_rules = {
+      mysql_vnet_rules = {
+        name                = "mysql-vnet-rule"
+      }
+    }
+        
+    mysql_database = {
```
Comment (LaurentLesle) [inline examples/mysql_server/101-vnet-rule-mysql/configuration.tfvars:65]: plural

File: examples/mysql_server/102-private-endpoint-mysql/configuration.tfvars [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    
+    extended_auditing_policy = {
+      storage_account = {
+        key = "auditing-re1"
+      }
+      retention_in_days = 7
+    }
+      
+    
+    mysql_configuration = {
```
Comment (LaurentLesle) [inline examples/mysql_server/102-private-endpoint-mysql/configuration.tfvars:41]: plural

File: examples/mysql_server/102-private-endpoint-mysql/configuration.tfvars [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+      mysql_configuration = {
+        name                = "interactive_timeout"
+        resource_group_name = "mysql_region1"
+        server_name         = "sales-re1"
+        value               = "600"
+      }
+    }
+         
+
+    mysql_database = {
```
Comment (LaurentLesle) [inline examples/mysql_server/102-private-endpoint-mysql/configuration.tfvars:51]: plural

File: examples/mysql_server/102-private-endpoint-mysql/configuration.tfvars [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    # Optional
+    private_endpoints = {
+      # Require enforce_private_link_endpoint_network_policies set to true on the subnet
+      private-link-level4 = {
+        name = "sales-mysql-re1"
+        remote_tfstate = {
+          tfstate_key = "foundations"
+          lz_key      = "launchpad"
+          output_key  = "vnets"
+        }
```
Comment (LaurentLesle) [inline examples/mysql_server/102-private-endpoint-mysql/configuration.tfvars:75]: remote_tstate is not required anymore. Plese delete the block

File: examples/mysql_server/103-private-endpoint-with-fw-rule-mysql/configuration.tfvars [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    mysql_configuration = {
+      mysql_configuration = {
+        name                = "interactive_timeout"
+        resource_group_name = "mysql_region1"
+        server_name         = "sales-re1"
+        value               = "600"
+      }
+    }
+    
+    mysql_database = {
```
Comment (LaurentLesle) [inline examples/mysql_server/103-private-endpoint-with-fw-rule-mysql/configuration.tfvars:59]: plural

File: examples/mysql_server/103-private-endpoint-with-fw-rule-mysql/configuration.tfvars [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+    # Optional
+    private_endpoints = {
+      # Require enforce_private_link_endpoint_network_policies set to true on the subnet
+      private-link-level4 = {
+        name = "sales-mysql-re1"
+        remote_tfstate = {
+          tfstate_key = "foundations"
+          lz_key      = "launchpad"
+          output_key  = "vnets"
+        }
```
Comment (LaurentLesle) [inline examples/mysql_server/103-private-endpoint-with-fw-rule-mysql/configuration.tfvars:82]: remote_tstate is not required anymore. Plese delete the block

File: examples/mysql_server/103-private-endpoint-with-fw-rule-mysql/configuration.tfvars [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+      name          = "mysql-vnet"
+      address_space = ["10.150.100.0/24"]
+      
+    }
+    #specialsubnets = {}
+    subnets = {
+      mysql_subnet = {
+        name    = "mysql_subnet"
+        cidr    = ["10.150.100.0/25"]
+        enforce_private_link_endpoint_network_policies = "true"
```
Comment (LaurentLesle) [inline examples/mysql_server/103-private-endpoint-with-fw-rule-mysql/configuration.tfvars:165]: make sure you run the terraform fmt --recursive /tf/caf

File: modules/databases/mysql_server/configuration.tf [hunk 11]
```diff
@@ -0,0 +1,12 @@
+# Configuration
+#
+
+resource "azurerm_mysql_configuration" "mysql_configuration" {
+  
+  for_each = var.settings.mysql_configuration
```
Comment (LaurentLesle) [inline modules/databases/mysql_server/configuration.tf:6]: plural (configurations)

File: modules/databases/mysql_server/database.tf [hunk 12]
```diff
@@ -0,0 +1,15 @@
+
+#
+# Firewall Rule
+#
+
+resource "azurerm_mysql_database" "mysql_database" {
+  
+  for_each = var.settings.mysql_database
```
Comment (LaurentLesle) [inline modules/databases/mysql_server/database.tf:8]: plural

**Review by LaurentLesle — CHANGES_REQUESTED** [review]

File: examples/mysql_server/101-vnet-rule-mysql/configuration.tfvars (hunk 3 above)
Comment (abhilash-keloth) [reply examples/mysql_server/101-vnet-rule-mysql/configuration.tfvars]: done

**Review by arnaudlh — APPROVED** [review]

**Pull request merged**